avocado/utils/process.py

```diff
@@ -267,16 +267,32 @@ class CmdResult(object):
     :param pid: ID of the process
     """
 
-    def __init__(self, command="", stdout="", stderr="",
+    def __init__(self, command="", stdout_bytes=b"", stderr_bytes=b"",
                  exit_status=None, duration=0, pid=None):
         self.command = command
         self.exit_status = exit_status
-        self.stdout = stdout
-        self.stderr = stderr
+        self.stdout_bytes = stdout_bytes
+        self.stderr_bytes = stderr_bytes
+        self._stdout = None
+        self._stderr = None
         self.duration = duration
         self.interrupted = False
         self.pid = pid
 
+    @property
+    def stdout(self):
+        if self._stdout is not None:
+            return self._stdout
+        self._stdout = self.stdout_bytes.decode('utf-8')
```

Comment:
I see, I'd not squash it as this way it's easier to follow the changes and it should not break things terribly...